src/main/kotlin/app/CodeLongevity.kt

```diff
@@ -0,0 +1,266 @@
+        println("Line '$text' - '${from.file}:${from.line}' added in $fc $fd")
+        println("  last known as '${to.file}:${to.line}' in $tc $td")
+    }
+}
+
+/**
+ * Used to compute age of code lines in the repo.
+ */
+class CodeLongevity(repoPath: String, tailRev: String) {
+    val repo = FileRepository(repoPath)
+    val head: RevCommit = RevWalk(repo).parseCommit(repo.resolve("HEAD"));
+    val tail: RevCommit? =
+        if (tailRev != "") RevWalk(repo).parseCommit(repo.resolve(tailRev))
+        else null
+
+    /**
+     * A list of all code lines, both alive and deleted, between the given
+     * revisions.
+     */
+    var codeLines: MutableList<CodeLine> = mutableListOf()
+
+    init {
+        compute()
+    }
+
+    // TODO(alex) debugging, remove it
+    fun ohNoDoesItReallyWork(email: String) {
+        var sum: Long = 0
+        var total: Long = 0;
+        for (line in codeLines) {
+            val author = line.from.commit.getAuthorIdent()
+            if (author.getEmailAddress() != email) {
+                continue
+            }
+            line.printme()
+            println("  Age: ${line.age} secs")
+            sum += line.age
+            total++
+        }
+
+        //println("All lines:")
+        //codeLines.forEach { line -> line.printme() }
+
+        var avg = if (total > 0) sum / total else 0
+        println("avg code line age for <$email> is ${avg} seconds, lines total: ${total}")
+    }
+
+    /**
+     * Scans through the repo for alive and deleted code lines, and stores them
+     * in the |codeLines| list.
+     */
+    private fun compute() {
+        val treeWalk = TreeWalk(repo)
+        treeWalk.setRecursive(true)
+        treeWalk.addTree(head.getTree())
+
+        val files: MutableMap<String, ArrayList<RevCommitLine>> = mutableMapOf()
+
+        // Build a map of file names and their code lines.
+        while (treeWalk.next()) {
+            val path = treeWalk.getPathString()
+            val lines = getFileLines(treeWalk.getObjectId(0), path, head)
+            if (lines != null) {
+                files.put(path, lines)
+            }
+        }
+  
+        val df = DiffFormatter(DisabledOutputStream.INSTANCE)
+        df.setRepository(repo)
+        df.setDetectRenames(true)
+
+        val revWalk = RevWalk(repo)
+        revWalk.markStart(head)
+
+        var commit: RevCommit? = revWalk.next()  // move the walker to the head
+        do {
+            var parentCommit: RevCommit? = revWalk.next()
+
+            println("commit: ${commit!!.getName()}; '${commit.getShortMessage()}'")
+            if (parentCommit != null) {
+                println("parent commit: ${parentCommit.getName()}; '${parentCommit.getShortMessage()}'")
+            }
+            else {
+                println("parent commit: null")
+            }
+
+            // A step back in commits history. Update the files map according
+            // to the diff.
+            val diffs = df.scan(parentCommit, commit)
```

Comment:
yes, that's the way to obtain a first commit changes / could you give an example?